libraries/standard/http/test/test-HTTPClient_ReadHeader.c

```diff
@@ -13,80 +32,109 @@ static const char * pTestResponse = "HTTP/1.1 200 OK"
                                     "test-header1: test-value1\r\n"
                                     "test-header2: test-value2\r\n"
                                     "\r\n";
-static const size_t headerFieldInRespLoc = 15;
-static const size_t headerFieldInRespLen = strlen( "test-header0" );
-static const size_t headerValInRespLoc = 29;
-static const size_t headerValInRespLen = strlen( "test-value0" );
+static const size_t headerFieldInRespLoc = 42;
+static const size_t headerFieldInRespLen = strlen( "test-header1" );
```

Comment:
Please use these instead:
```
static const size_t headerFieldInRespLen = sizeof( "test-header1" ) - 1u;
...
static const size_t headerValInRespLen = sizeof( "test-value1" ) - 1u;
```

Reply:
What's the reason of preferring `sizeof` over `strlen` in a test file? We have control over the test response used in the test file to be confident that it is a constant string...

Reply:
Using strlen() here is a bit weird, I thought global static variables needed to be initialized with expressions known at compile time. This also gets rid of the warning spit out.

Reply:
Good point about using compile-time constants. (In C++, they use a separate type specifier `constexpr` to represent compile-time computable expressions). Changed to `sizeof()`